### ISSUE TITLE
Command line option to start auto-type

### ISSUE THREAD

**Issue by adrian-amaglio** [keepassxreboot/keepassxc#8238]
## Summary
I am using a software (sxhkd) to manage all my shortcuts in one place.
In this configuration, it would be nice to have the keepassxc auto-type shortcut in this same place.
For that, one could use a command line option that trigger the auto-type process (ask for password if needed, select the entry to auto-type…)

## Examples
This command line could start the auto-type process:
``` $ keepassxc --auto-type ```

## Context
This feature keeps the jobs separated. Keepassxc manages passwords, sxhkd manages the shortcuts.

## Security
This may be a security concern but I can’t tell myself since I do not know how keepassxc handle key shortcuts.

**Comment by Daklon** [comment]
I'm also interested on this. I'm trying to automate the login on some services that requires some commands to be executed first, then a browser window will popup and the credentials must be set there. Having the possibility to trigger the autotype from the cli will be very helpful to automate the login process

**Comment by droidmonkey** [comment]
Why doesn't the global keyboard shortcut work?

**Comment by Daklon** [comment]
Hi @droidmonkey, I tried your approach, using the global keyboard shortcut and it's an improvement, but still not 100% automated. When I trigger the global keyboard shortcut a popup window appears to let me select which credential I want to use. I've tried renaming the entry on keepassxc as the window name and it helped filtering but the popup dialog still appears. Are you aware if there is any way to just use the single credential that appears automatically?

**Comment by defini-tiv** [comment]
@droidmonkey 
> Why doesn't the global keyboard shortcut work?

1. It doesn't work on Wayland (i know there are other problems, but there may be compositors that don't support global keyboard shortcuts for longer than the other problems persist.)  
2. Making it better scriptable
3. Having the ability to control all shortcuts via one daemon is preferred by many people, its following the unix-philosophy and can be advantageous for multiple reasons.

**Comment by droidmonkey** [comment]
Auto-Type doesn't work on Wayland at all

**Comment by defini-tiv** [comment]
Yes, im aware there are other hurdles. But it is targeted to get these resolved in the future, right?
So having a secondary way to access auto-type may be helpfull for ensuring global keyboard shortcuts work anywhere (e.g. even in exotic compositors) in addition to beeing beneficial in other ways. (As i wrote already above)

**Comment by droidmonkey** [comment]
Fundamental features need to be built into the Wayland protocol and deployed across distributions before we can even begin work on Wayland Auto-Type. That is not under our control. Global shortcuts should also be one of those features. Either that or freedesktop.org figures out a common means to implement them, like with dbus or something. Linux is such a hot mess.

**Comment by defini-tiv** [comment]
With the feature from this request we don't *need* any global shortcut feature. The other problems are not unsolveable eighter.

#### But to be clear:
If wayland support would be the only selling point of this, of crouse it would be stupid to borther and implement it now. But i don't see why it shoulden count as a plus point when there are other usecases.

#### Wayland is not the *reason* to implent this, it is a good *example* of such a feature beeing useful.

**Comment by droidmonkey** [comment]
I never said it was the only reason

**Comment by matkoniecz** [comment]
> Why doesn't the global keyboard shortcut work?

`keepass2 -auto-type` was smart enough to start keepass if it is not running already

with in-program global shortcut I think I need it running already for it to be able to capture it (so I would need to figure out how to add it to autostart, and start in minimized/tray mode)

**Comment by matkoniecz** [comment]
> With the feature from this request we don't need any global shortcut feature. 

not so sure, for many people dealing with CLI invocations in global OS shortcuts may be too complex

**Comment by KuttKatrea** [comment]
Another use case for this: in Windows I can't map Win+Shift+A on KeePassXC directly.

Previously I was using `keepass2 -auto-type` with AutoHotKey for this same scenario.
